src/helpers/figures/charts/pie_chart.ts

```diff
@@ -260,6 +261,53 @@ function calculatePercentage(
   return percentage.toFixed(2);
 }
 
+function filterNegativeValues(
+  labels: readonly string[],
+  datasets: readonly DatasetValues[]
+): { labels: string[]; dataSetsValues: DatasetValues[] } {
+  if (datasets.length === 1) {
+    const filteredData: number[] = [];
+    const filteredLabels: string[] = [];
+
+    for (let i = 0; i < labels.length; i++) {
+      const label = labels[i];
```

Comment:
what if labels.length < datasets[0].length , you do not change some change datasets values.
I think you should jave the same approach with `numberOfDataPoints` which leads me to believe that there might not be a need to do 2 different cases depending on the number of datasets.